<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1003</width>
    <height>500</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <family>微软雅黑</family>
   </font>
  </property>
  <property name="windowTitle">
   <string/>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QSplitter" name="splitter_hori">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QTabWidget" name="tabWidget">
       <property name="currentIndex">
        <number>0</number>
       </property>
       <widget class="FileWidget" name="fileWidget">
        <attribute name="title">
         <string>合约项目</string>
        </attribute>
       </widget>
       <widget class="ContractWidget" name="contractWidget">
        <attribute name="title">
         <string>已注册合约</string>
        </attribute>
       </widget>
      </widget>
      <widget class="QSplitter" name="splitter_ver">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <widget class="ContextWidget" name="contentWidget" native="true">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
       </widget>
       <widget class="OutputWidget" name="outputWidget" native="true"/>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1003</width>
     <height>35</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>0</width>
     <height>35</height>
    </size>
   </property>
   <widget class="QMenu" name="fileMenu">
    <property name="title">
     <string>文件</string>
    </property>
    <widget class="QMenu" name="menu_new">
     <property name="title">
      <string>新建文件</string>
     </property>
     <addaction name="newContractAction_glua"/>
     <addaction name="newContractAction_java"/>
     <addaction name="newContractAction_csharp"/>
     <addaction name="newContractAtion_kotlin"/>
    </widget>
    <widget class="QMenu" name="menu">
     <property name="title">
      <string>导入文件</string>
     </property>
     <addaction name="importContractAction_glua"/>
     <addaction name="importContractAction_java"/>
     <addaction name="importContractAction_csharp"/>
     <addaction name="importContractAction_kotlin"/>
    </widget>
    <addaction name="menu_new"/>
    <addaction name="menu"/>
    <addaction name="separator"/>
    <addaction name="saveAction"/>
    <addaction name="savaAsAction"/>
    <addaction name="saveAllAction"/>
    <addaction name="separator"/>
    <addaction name="closeAction"/>
    <addaction name="closeAllAction"/>
    <addaction name="separator"/>
    <addaction name="configAction"/>
    <addaction name="separator"/>
    <addaction name="exitAction"/>
   </widget>
   <widget class="QMenu" name="contractMenu">
    <property name="title">
     <string>合约</string>
    </property>
    <addaction name="registerAction"/>
    <addaction name="upgradeAction"/>
    <addaction name="callAction"/>
    <addaction name="transferAction"/>
    <addaction name="withdrawAction"/>
    <addaction name="separator"/>
    <addaction name="importAction"/>
    <addaction name="exportAction"/>
   </widget>
   <widget class="QMenu" name="setMenu">
    <property name="title">
     <string>工具</string>
    </property>
    <addaction name="changeChainAction"/>
    <addaction name="consoleAction"/>
    <addaction name="accountListAction"/>
    <addaction name="transferToAccountAction"/>
   </widget>
   <widget class="QMenu" name="helpMenu">
    <property name="title">
     <string>帮助</string>
    </property>
    <addaction name="helpAction"/>
    <addaction name="aboutAction"/>
   </widget>
   <widget class="QMenu" name="editMenu">
    <property name="title">
     <string>编辑</string>
    </property>
    <addaction name="undoAction"/>
    <addaction name="redoAction"/>
   </widget>
   <widget class="QMenu" name="debugMenu">
    <property name="title">
     <string>调试</string>
    </property>
    <addaction name="compileAction"/>
    <addaction name="separator"/>
    <addaction name="debugAction"/>
    <addaction name="stepAction"/>
    <addaction name="stopAction"/>
    <addaction name="separator"/>
    <addaction name="TabBreaPointAction"/>
    <addaction name="DeleteAllBreakpointAction"/>
   </widget>
   <addaction name="fileMenu"/>
   <addaction name="editMenu"/>
   <addaction name="contractMenu"/>
   <addaction name="debugMenu"/>
   <addaction name="setMenu"/>
   <addaction name="helpMenu"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <property name="minimumSize">
    <size>
     <width>0</width>
     <height>35</height>
    </size>
   </property>
   <property name="movable">
    <bool>false</bool>
   </property>
   <property name="allowedAreas">
    <set>Qt::TopToolBarArea</set>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="newContractAction_glua"/>
   <addaction name="saveAction"/>
   <addaction name="saveAllAction"/>
   <addaction name="separator"/>
   <addaction name="undoAction"/>
   <addaction name="redoAction"/>
   <addaction name="separator"/>
   <addaction name="registerAction"/>
   <addaction name="upgradeAction"/>
   <addaction name="callAction"/>
   <addaction name="transferAction"/>
   <addaction name="withdrawAction"/>
   <addaction name="importAction"/>
   <addaction name="exportAction"/>
   <addaction name="separator"/>
   <addaction name="compileAction"/>
   <addaction name="debugAction"/>
   <addaction name="stepAction"/>
   <addaction name="stopAction"/>
   <addaction name="separator"/>
   <addaction name="changeChainAction"/>
   <addaction name="consoleAction"/>
   <addaction name="accountListAction"/>
   <addaction name="transferToAccountAction"/>
   <addaction name="separator"/>
   <addaction name="helpAction"/>
  </widget>
  <action name="newContractAction_glua">
   <property name="icon">
    <iconset resource="chain.qrc">
     <normaloff>:/pic/newContract_enable.png</normaloff>
     <disabledoff>:/pic/newContract_disable.png</disabledoff>:/pic/newContract_enable.png</iconset>
   </property>
   <property name="text">
    <string>新建uvlua合约</string>
   </property>
  </action>
  <action name="newContractAction_csharp">
   <property name="icon">
    <iconset resource="chain.qrc">
     <normaloff>:/pic/newContract_enable.png</normaloff>
     <disabledoff>:/pic/newContract_disable.png</disabledoff>:/pic/newContract_enable.png</iconset>
   </property>
   <property name="text">
    <string>新建c#合约</string>
   </property>
  </action>
  <action name="newContractAction_java">
   <property name="icon">
    <iconset resource="chain.qrc">
     <normaloff>:/pic/newContract_enable.png</normaloff>
     <disabledoff>:/pic/newContract_disable.png</disabledoff>:/pic/newContract_enable.png</iconset>
   </property>
   <property name="text">
    <string>新建java合约</string>
   </property>
  </action>
  <action name="newContractAtion_kotlin">
   <property name="icon">
    <iconset resource="chain.qrc">
     <normaloff>:/pic/newContract_enable.png</normaloff>
     <disabledoff>:/pic/newContract_disable.png</disabledoff>:/pic/newContract_enable.png</iconset>
   </property>
   <property name="text">
    <string>新建kotlin合约</string>
   </property>
  </action>
  <action name="saveAction">
   <property name="icon">
    <iconset resource="chain.qrc">
     <normaloff>:/pic/save_enable.png</normaloff>
     <disabledoff>:/pic/save_disable.png</disabledoff>:/pic/save_enable.png</iconset>
   </property>
   <property name="text">
    <string>保存</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="savaAsAction">
   <property name="text">
    <string>另存为</string>
   </property>
  </action>
  <action name="saveAllAction">
   <property name="icon">
    <iconset resource="chain.qrc">
     <normaloff>:/pic/saveAll_enable.png</normaloff>
     <disabledoff>:/pic/saveAll_disable.png</disabledoff>:/pic/saveAll_enable.png</iconset>
   </property>
   <property name="text">
    <string>保存全部</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="compileAction">
   <property name="icon">
    <iconset resource="chain.qrc">
     <normaloff>:/pic/compile_enable.png</normaloff>
     <disabledoff>:/pic/compile_disable.png</disabledoff>:/pic/compile_enable.png</iconset>
   </property>
   <property name="text">
    <string>编译</string>
   </property>
   <property name="shortcut">
    <string>F7</string>
   </property>
  </action>
  <action name="closeAction">
   <property name="text">
    <string>关闭</string>
   </property>
  </action>
  <action name="exitAction">
   <property name="text">
    <string>退出</string>
   </property>
  </action>
  <action name="undoAction">
   <property name="icon">
    <iconset resource="chain.qrc">
     <normaloff>:/pic/undo_enable.png</normaloff>
     <disabledoff>:/pic/undo_disable.png</disabledoff>:/pic/undo_enable.png</iconset>
   </property>
   <property name="text">
    <string>撤销</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
  </action>
  <action name="redoAction">
   <property name="icon">
    <iconset resource="chain.qrc">
     <normaloff>:/pic/redo_enable.png</normaloff>
     <disabledoff>:/pic/redo_disable.png</disabledoff>:/pic/redo_enable.png</iconset>
   </property>
   <property name="text">
    <string>恢复</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Y</string>
   </property>
  </action>
  <action name="importAction">
   <property name="icon">
    <iconset resource="chain.qrc">
     <normaloff>:/pic/import_enable.png</normaloff>
     <disabledoff>:/pic/import_disable.png</disabledoff>:/pic/import_enable.png</iconset>
   </property>
   <property name="text">
    <string>导入合约</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="exportAction">
   <property name="icon">
    <iconset resource="chain.qrc">
     <normaloff>:/pic/export_enable.png</normaloff>
     <disabledoff>:/pic/export_disable.png</disabledoff>:/pic/export_enable.png</iconset>
   </property>
   <property name="text">
    <string>导出合约</string>
   </property>
   <property name="toolTip">
    <string>合约导出</string>
   </property>
  </action>
  <action name="registerAction">
   <property name="icon">
    <iconset resource="chain.qrc">
     <normaloff>:/pic/registerContract_enable.png</normaloff>
     <disabledoff>:/pic/registerContract_disable.png</disabledoff>:/pic/registerContract_enable.png</iconset>
   </property>
   <property name="text">
    <string>注册合约</string>
   </property>
  </action>
  <action name="transferAction">
   <property name="icon">
    <iconset resource="chain.qrc">
     <normaloff>:/pic/tranferToContract_enable.png</normaloff>
     <disabledoff>:/pic/tranferToContract_disable.png</disabledoff>:/pic/tranferToContract_enable.png</iconset>
   </property>
   <property name="text">
    <string>合约充值</string>
   </property>
   <property name="toolTip">
    <string>充值</string>
   </property>
  </action>
  <action name="callAction">
   <property name="icon">
    <iconset resource="chain.qrc">
     <normaloff>:/pic/callContract_enable.png</normaloff>
     <disabledoff>:/pic/callContract_disable.png</disabledoff>:/pic/callContract_enable.png</iconset>
   </property>
   <property name="text">
    <string>调用合约</string>
   </property>
  </action>
  <action name="upgradeAction">
   <property name="icon">
    <iconset resource="chain.qrc">
     <normaloff>:/pic/upgrade_enable.png</normaloff>
     <disabledoff>:/pic/upgrade_disable.png</disabledoff>:/pic/upgrade_enable.png</iconset>
   </property>
   <property name="text">
    <string>升级合约</string>
   </property>
  </action>
  <action name="withdrawAction">
   <property name="icon">
    <iconset resource="chain.qrc">
     <normaloff>:/pic/withdrawContract_enable.png</normaloff>
     <disabledoff>:/pic/withdrawContract_disable.png</disabledoff>:/pic/withdrawContract_enable.png</iconset>
   </property>
   <property name="text">
    <string>销毁合约</string>
   </property>
  </action>
  <action name="helpAction">
   <property name="icon">
    <iconset resource="chain.qrc">
     <normaloff>:/pic/help_enable.png</normaloff>
     <disabledoff>:/pic/help_disable.png</disabledoff>:/pic/help_enable.png</iconset>
   </property>
   <property name="text">
    <string>查看帮助</string>
   </property>
   <property name="shortcut">
    <string>F1</string>
   </property>
  </action>
  <action name="aboutAction">
   <property name="text">
    <string>关于</string>
   </property>
  </action>
  <action name="changeChainAction">
   <property name="icon">
    <iconset resource="chain.qrc">
     <normaloff>:/pic/changeToFormal_enable.png</normaloff>
     <disabledoff>:/pic/changeToFormal_disable.png</disabledoff>:/pic/changeToFormal_enable.png</iconset>
   </property>
   <property name="text">
    <string>切换到正式链</string>
   </property>
  </action>
  <action name="enterSandboxAction">
   <property name="icon">
    <iconset resource="chain.qrc">
     <normaloff>:/pic/enterSandbox_enable.png</normaloff>
     <disabledoff>:/pic/enterSandbox_disable.png</disabledoff>:/pic/enterSandbox_enable.png</iconset>
   </property>
   <property name="text">
    <string>沙盒模式</string>
   </property>
  </action>
  <action name="exitSandboxAction">
   <property name="text">
    <string>退出沙盒模式</string>
   </property>
  </action>
  <action name="accountListAction">
   <property name="icon">
    <iconset resource="chain.qrc">
     <normaloff>:/pic/account_enable.png</normaloff>
     <disabledoff>:/pic/account_disable.png</disabledoff>:/pic/account_enable.png</iconset>
   </property>
   <property name="text">
    <string>账户管理</string>
   </property>
  </action>
  <action name="consoleAction">
   <property name="icon">
    <iconset resource="chain.qrc">
     <normaloff>:/pic/console_enable.png</normaloff>
     <disabledoff>:/pic/console_disable.png</disabledoff>:/pic/console_enable.png</iconset>
   </property>
   <property name="text">
    <string>控制台</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+D</string>
   </property>
  </action>
  <action name="transferToAccountAction">
   <property name="icon">
    <iconset resource="chain.qrc">
     <normaloff>:/pic/transfer_enable.png</normaloff>
     <disabledoff>:/pic/transfer_disable.png</disabledoff>:/pic/transfer_enable.png</iconset>
   </property>
   <property name="text">
    <string>转账</string>
   </property>
  </action>
  <action name="closeAllAction">
   <property name="text">
    <string>全部关闭</string>
   </property>
  </action>
  <action name="debugAction">
   <property name="icon">
    <iconset resource="chain.qrc">
     <normaloff>:/pic/startDebug_enable.png</normaloff>
     <disabledoff>:/pic/startDebug_disable.png</disabledoff>:/pic/startDebug_enable.png</iconset>
   </property>
   <property name="text">
    <string>开始调试</string>
   </property>
   <property name="shortcut">
    <string>F5</string>
   </property>
  </action>
  <action name="stopAction">
   <property name="icon">
    <iconset resource="chain.qrc">
     <normaloff>:/pic/stop_enable.png</normaloff>
     <disabledoff>:/pic/stop_disable.png</disabledoff>:/pic/stop_enable.png</iconset>
   </property>
   <property name="text">
    <string>停止调试</string>
   </property>
  </action>
  <action name="stepAction">
   <property name="icon">
    <iconset resource="chain.qrc">
     <normaloff>:/pic/step_enable.png</normaloff>
     <disabledoff>:/pic/step_disable.png</disabledoff>:/pic/step_enable.png</iconset>
   </property>
   <property name="text">
    <string>逐步</string>
   </property>
  </action>
  <action name="importContractAction_glua">
   <property name="icon">
    <iconset resource="chain.qrc">
     <normaloff>:/pic/import_enable.png</normaloff>
     <disabledoff>:/pic/import_disable.png</disabledoff>:/pic/import_enable.png</iconset>
   </property>
   <property name="text">
    <string>导入uvlua合约</string>
   </property>
  </action>
  <action name="importContractAction_java">
   <property name="icon">
    <iconset resource="chain.qrc">
     <normaloff>:/pic/import_enable.png</normaloff>
     <disabledoff>:/pic/import_disable.png</disabledoff>:/pic/import_enable.png</iconset>
   </property>
   <property name="text">
    <string>导入java合约</string>
   </property>
  </action>
  <action name="importContractAction_csharp">
   <property name="icon">
    <iconset resource="chain.qrc">
     <normaloff>:/pic/import_enable.png</normaloff>
     <disabledoff>:/pic/import_disable.png</disabledoff>:/pic/import_enable.png</iconset>
   </property>
   <property name="text">
    <string>导入c#合约</string>
   </property>
  </action>
  <action name="importContractAction_kotlin">
   <property name="icon">
    <iconset resource="chain.qrc">
     <normaloff>:/pic/import_enable.png</normaloff>
     <disabledoff>:/pic/import_disable.png</disabledoff>:/pic/import_enable.png</iconset>
   </property>
   <property name="text">
    <string>导入kotlin合约</string>
   </property>
  </action>
  <action name="configAction">
   <property name="icon">
    <iconset resource="chain.qrc">
     <normaloff>:/pic/configure_enable.png</normaloff>:/pic/configure_enable.png</iconset>
   </property>
   <property name="text">
    <string>配置</string>
   </property>
  </action>
  <action name="TabBreaPointAction">
   <property name="text">
    <string>切换断点</string>
   </property>
   <property name="shortcut">
    <string>F9</string>
   </property>
  </action>
  <action name="DeleteAllBreakpointAction">
   <property name="text">
    <string>删除所有断点</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+F9</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>FileWidget</class>
   <extends>QWidget</extends>
   <header>filewidget/filewidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ContextWidget</class>
   <extends>QWidget</extends>
   <header>contentwidget/contextwidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>OutputWidget</class>
   <extends>QWidget</extends>
   <header>outputwidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ContractWidget</class>
   <extends>QWidget</extends>
   <header>contractwidget/contractwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="chain.qrc"/>
 </resources>
 <connections/>
</ui>
